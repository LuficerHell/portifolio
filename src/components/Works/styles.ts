import styled from 'styled-components'

export const Container = styled.div`
  box-shadow: 0 4px 25px rgba(14, 36, 49, 0.15);
  border-radius: 0.5rem;
  overflow: hidden;
`

export const WorkImage = styled.img`
  transition: 1s;
  cursor: pointer;

  :hover {
    transform: scale(1.1);
  }
`
