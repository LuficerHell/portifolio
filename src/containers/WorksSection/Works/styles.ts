import styled from 'styled-components'

export const Container = styled.div`
  box-shadow: 0 4px 25px rgba(14, 36, 49, 0.15);
  border-radius: 0.5rem;
  overflow: hidden;
`

export const WorkImage = styled.img`
  transition: 1.2s;
  cursor: pointer;

  :hover {
    transform: scale(1.2);
  }
`
