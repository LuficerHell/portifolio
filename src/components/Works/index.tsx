import * as S from './Works.styles';

interface ILinkImage {
  imageLink: string;
  imageDescription: string;
}

const Works = ({ imageLink, imageDescription }: ILinkImage) => {
  return (
    <S.Container>
      <S.WorkImage src={imageLink} alt={imageDescription} />
    </S.Container>
  );
};

export default Works;
