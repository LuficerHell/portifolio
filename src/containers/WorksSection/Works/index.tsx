import * as S from './styles'

interface ILinkImage {
  imageLink: string;
  imageDescription: string;
}

const Works = ( { imageLink, imageDescription }: ILinkImage ) => {
  return (
    <S.Container>
      <S.WorkImage src={imageLink} alt={imageDescription} />
    </S.Container>
  )
}

export default Works
