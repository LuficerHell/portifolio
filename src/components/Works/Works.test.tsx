import { screen, render } from '@testing-library/react';

import Works from './';

describe('<Works />', () => {
  it('should render the heading', () => {
    const { container } = render(<Works />);

    // expect(
    //   screen.getByRole('heading', { name: /Works/i }),
    // ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
