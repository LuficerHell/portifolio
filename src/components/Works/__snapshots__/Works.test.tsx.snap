// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Works /> should render the heading 1`] = `
.c0 {
  box-shadow: 0 4px 25px rgba(14,36,49,0.15);
  border-radius: 0.5rem;
  overflow: hidden;
}

.c1 {
  -webkit-transition: 1s;
  transition: 1s;
  cursor: pointer;
}

.c1:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div
  class="c0"
>
  <img
    class="c1"
  />
</div>
`;
